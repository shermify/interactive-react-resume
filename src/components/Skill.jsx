import React from 'react';
import PropTypes from 'prop-types';
import defaultImage from '../images/logos/defaultSkill.png';

const Skill = props => {
  const { size, percent, label, img } = props;
  return (
    <div className="skill">
      <div className="row">
        <div className="col">
          <img src={img} alt={label} height="45px" width="45px" />
        </div>
        <div className="col">
          <svg width={size} height="15px">
            <defs>
              <linearGradient id="Gradient1">
                <stop stopColor="blue" offset="0%" />
                <stop stopColor="purple" offset="100%" />
              </linearGradient>
            </defs>
            <rect x="0" y="0" width="100" height="50" fill="lightgray" />
            <rect x="0" y="0" width={percent} height="50" fill="url(#Gradient1)" />
          </svg>
          <p style={{ width: size }}>{label}</p>
        </div>
      </div>
    </div>
  );
};

Skill.propTypes = {
  percent: PropTypes.number, // value the chart should show
  label: PropTypes.string, // label for the chart
  size: PropTypes.number, // diameter of chart
  img: PropTypes.string,
};
Skill.defaultProps = {
  percent: 0,
  label: '',
  size: 116,
  img: defaultImage,
};

export default Skill;
