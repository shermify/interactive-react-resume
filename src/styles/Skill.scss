.skill {
  display: inline-block;
  margin: 2%;
  text-align: left;

  .col {
    padding-left: 5px;
    padding-right: 5px;
  }
}
