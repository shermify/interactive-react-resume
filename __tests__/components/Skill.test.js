import React from 'react';
import renderer from 'react-test-renderer';
import Skill from '../../src/components/Skill';

describe('<Skill />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Skill
        label="Test skill"
        percent={60}
        size={100}
        strokeWidth={20}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
