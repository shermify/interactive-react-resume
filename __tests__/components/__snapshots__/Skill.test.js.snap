// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skill /> renders correctly 1`] = `
<div
  className="skill"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <img
        alt="Test skill"
        height="45px"
        src="test-file-stub"
        width="45px"
      />
    </div>
    <div
      className="col"
    >
      <svg
        height="15px"
        width={100}
      >
        <defs>
          <linearGradient
            id="Gradient1"
          >
            <stop
              offset="0%"
              stopColor="blue"
            />
            <stop
              offset="100%"
              stopColor="purple"
            />
          </linearGradient>
        </defs>
        <rect
          fill="lightgray"
          height="50"
          width="100"
          x="0"
          y="0"
        />
        <rect
          fill="url(#Gradient1)"
          height="50"
          width={60}
          x="0"
          y="0"
        />
      </svg>
      <p
        style={
          Object {
            "width": 100,
          }
        }
      >
        Test skill
      </p>
    </div>
  </div>
</div>
`;
